File: Etap 5 - Implementacja C#/source code/MiddleAges/army_classes/Army.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MiddleAges
{
    class Army : Price
    {
        //#################################################
        // CONSTRUCTORS
        //#################################################

        //#################################################
        // MEMBER FUNCTIONS
        //#################################################
        public void AddQuantity(int value)
        {
            _quantity += value;
        }

        public void RemoveQuantity(int value)
        {
            _quantity -= value;
        }

        public void AddPower(int value)
        {
            _power += value;
        }

        public void RemovePower(int value)
        {
            _power -= value;
        }

        public void MultiplyPower(double value)
        {
            _power *= value;
        }

        //#################################################
        // GETTERS & SETTERS
        //#################################################
        public double Power
        {
            get { return _power;  }
            set { _power = value; }
        }

        public int Quantity
        {
            get { return _quantity; }
            set { _quantity = value; }
        }

        public int Level
        {
            get { return _level; }
            set { _level = value; }
        }

        //#################################################
        // MEMBER VARIABLES
        //#################################################
        protected double _power;
        protected int _quantity;
        protected int _level;

    }
}

File: Etap 4 - Implementacja C++/MiddleAges/army.cpp
#include "army.h"

//#################################################
// CONSTRUCTORS
//#################################################
Army::Army()
{

}

//#################################################
// MEMBER FUNCTIONS
//#################################################
void Army::addQuantity(int value)
{
    quantity += value;
}

void Army::removeQuantity(int value)
{
    quantity -= value;
}

void Army::addPower(int value)
{
    power += value;
}

void Army::removePower(int value)
{
    power -= value;
}

void Army::multiplyPower(double value)
{
    power *= value;
}

//#################################################
// SETTERS
//#################################################
void Army::setPower(int value)
{
    power = value;
}

void Army::setQuantity(int value)
{
    quantity = value;
}

void Army::setLevel(int value)
{
    level = value;
}

//#################################################
// GETTERS
//#################################################
double Army::getPower()
{
    return power;
}

int Army::getLevel()
{
    return level;
}

int Army::getQuantity()
{
    return quantity;
}
